cmd/zhack/zhack.c

```diff
@@ -133,8 +133,10 @@ import_pool(const char *target, boolean_t readonly)
 	char *name;
 
 	kernel_init(readonly ? FREAD : (FREAD | FWRITE));
-	g_zfs = libzfs_init();
-	ASSERT(g_zfs != NULL);
+	if ((g_zfs = libzfs_init()) == NULL) {
+		(void) fprintf(stderr, "%s", libzfs_error_init(errno));
+		return;
```

Comment:
Shouldn't a `libzfs_init()` failure here be fatal, especially since you've removed the `ASSERT`?

Now if `libzfs_init()` fails, you return early from `import_pool()` . That function returns a void, so you can't return the error to the caller (`zhack_spa_open()`) which obliviously rumbles on.